### PR TITLE
Add  groupby SUM_OF_SQUARES support

### LINKED ISSUE #4667: [FEA] `SUM_OF_SQUARES` aggregation for groupby in libcudf
It would be great if libcudf groupby could provide a `SUM_OF_SQUARES` aggregation.

**Additional context**
While Pandas does not have a `sum_of_squares` aggregation explicitly, it looks like the `dask_cudf` implementation of `groupby.std` could benefit from a `sum_of_squares` aggregation greatly (cc: @quasiben).

### PR DESCRIPTION
**Pull request by karthikeyann**
closes #4667
- Add hash groupby SUM_OF_SQUARES support
- considering ptxas 10.2 crash, it's enabled for floating types only in unit tests.

### CODE REVIEW

**Review by codereport — APPROVED** [review]
lgtm :+1:

**Review by kkraus14 — APPROVED** [review]
cmake lgtm

File: cpp/include/cudf/detail/aggregation/aggregation.cuh [hunk 1]
```diff
@@ -314,6 +314,32 @@ struct update_target_element<dictionary32, aggregation::SUM, target_has_nulls, s
...
+  __device__ void operator()(mutable_column_device_view target,
+                             size_type target_index,
+                             column_device_view source,
+                             size_type source_index) const noexcept
+  {
+// This code will segfault in nvcc/ptxas 10.2 only
+// https://nvbugswb.nvidia.com/NvBugs5/SWBug.aspx?bugid=3186317
+// Enabling only for 2 types does not segfault. Using for unit tests.
+#if (__CUDACC_VER_MAJOR__ == 10) and (__CUDACC_VER_MINOR__ == 2)
+    if (not std::is_floating_point<Source>::value) return;
```
Comment (jrhemstad) [inline cpp/include/cudf/detail/aggregation/aggregation.cuh:332]: So will this just yield silently incorrect results on 10.2? I think we should detect this and throw.

Comment (karthikeyann) [reply cpp/include/cudf/detail/aggregation/aggregation.cuh]: Updated. modified to throw.

**Review by jrhemstad — APPROVED** [review]

**Comment by karthikeyann** [comment]
@gpucibot merge

**Pull request merged**